examples/stress_tests/many_materials.rs

```diff
@@ -0,0 +1,121 @@
+//! Benchmark to test rendering many animated materials
+
+use bevy::{
+    diagnostic::{FrameTimeDiagnosticsPlugin, LogDiagnosticsPlugin},
+    prelude::*,
+    utils::HashSet,
+    window::{PresentMode, WindowPlugin, WindowResolution},
+};
+use std::f32::consts::PI;
+
+#[derive(Component)]
+struct Floor;
+
+fn main() {
+    App::new()
+        .add_plugins((
+            DefaultPlugins.set(WindowPlugin {
+                primary_window: Some(Window {
+                    resolution: WindowResolution::new(1920.0, 1080.0)
+                        .with_scale_factor_override(1.0),
+                    title: "many_materials".into(),
+                    present_mode: PresentMode::AutoNoVsync,
+                    ..default()
+                }),
+                ..default()
+            }),
+            FrameTimeDiagnosticsPlugin,
+            LogDiagnosticsPlugin::default(),
+        ))
+        .add_systems(Startup, setup)
+        .add_systems(Update, (animate_materials, make_materials_unique))
+        .run();
+}
+
+fn setup(
+    mut commands: Commands,
+    asset_server: Res<AssetServer>,
+    mut meshes: ResMut<Assets<Mesh>>,
+    mut materials: ResMut<Assets<StandardMaterial>>,
+) {
+    let n = 4;
+
+    // Camera
+    commands.spawn(Camera3dBundle {
+        transform: Transform::from_xyz(n as f32 + 1.0, 1.0, n as f32 + 1.0)
+            .looking_at(Vec3::new(0.0, -0.5, 0.0), Vec3::Y),
+        ..default()
+    });
+
+    // Plane
+    commands.spawn((
+        PbrBundle {
+            mesh: meshes.add(shape::Plane::from_size(50.0)),
+            material: materials.add(Color::rgb(0.3, 0.5, 0.3)),
+            ..default()
+        },
+        Floor,
+    ));
+
+    // Light
+    commands.spawn(DirectionalLightBundle {
+        transform: Transform::from_rotation(Quat::from_euler(EulerRot::ZYX, 0.0, 1.0, -PI / 4.)),
+        directional_light: DirectionalLight {
+            illuminance: 3000.0,
+            shadows_enabled: true,
+            ..default()
+        },
+        ..default()
+    });
+
+    // Helmets
+    let helmet = asset_server.load("models/FlightHelmet/FlightHelmet.gltf#Scene0");
```

Comment:
using the helmet instead of a pbrbundle means you're also adding hierarchy and complex meshes to the stress test

it also means we don't have direct control of the material and can't try easily if it's because one of the property of it

I would prefer to use a `PbrBundle` as it makes a more focused stress test, and it wouldn't need the `make_materials_unique` system

Reply:
Hmm, I think I agree with this analysis. It would be better to be able to properly isolate the effects here.